From: Armin Novak <armin.novak@thincast.com>
Date: Tue, 20 Jul 2021 10:57:28 +0200
Subject: Fixed FFMPEG detection without pkgconfig

(cherry picked from commit 2c31c367598863d73c3c5cc3a8b93bfebe2dfb60)
(cherry picked from commit 2b65b7cb123cc5c9950aa1bf4a136dae63349951)
---
 cmake/FindFFmpeg.cmake | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/cmake/FindFFmpeg.cmake b/cmake/FindFFmpeg.cmake
index 78d9bf6..ffa1641 100644
--- a/cmake/FindFFmpeg.cmake
+++ b/cmake/FindFFmpeg.cmake
@@ -27,12 +27,20 @@ find_library(AVUTIL_LIBRARY avutil PATHS ${AVUTIL_LIBRARY_DIRS})
 
 # swresample
 find_path(SWRESAMPLE_INCLUDE_DIR libswresample/swresample.h PATHS ${SWRESAMPLE_INCLUDE_DIRS})
-find_library(SWRESAMPLE_LIBRARY swresample PATHS ${SWRESAMPLE_LIBRARY_DIRS})
+find_library(SWRESAMPLE_LIBRARY NAMES swresample swresample-3 PATHS ${SWRESAMPLE_LIBRARY_DIRS})
+
+if (SWRESAMPLE_INCLUDE_DIR AND SWRESAMPLE_LIBRARY)
+	set(SWRESAMPLE_FOUND ON)
+endif()
 
 # avresample
 find_path(AVRESAMPLE_INCLUDE_DIR libavresample/avresample.h PATHS ${AVRESAMPLE_INCLUDE_DIRS})
 find_library(AVRESAMPLE_LIBRARY avresample PATHS ${AVRESAMPLE_LIBRARY_DIRS})
 
+if (AVRESAMPLE_INCLUDE_DIR AND AVRESAMPLE_LIBRARY)
+	set(AVRESAMPLE_FOUND ON)
+endif()
+
 if (AVCODEC_INCLUDE_DIR AND AVCODEC_LIBRARY)
 	set(AVCODEC_FOUND TRUE)
 endif(AVCODEC_INCLUDE_DIR AND AVCODEC_LIBRARY)
